### ISSUE TITLE
Illegal instruction (core dumped)

### ISSUE THREAD

**Issue by Chida82** [madMAx43v3r/chia-plotter#88]
Hi,
in an 10 year's old server I have this error in the same point

![image](https://user-images.githubusercontent.com/6495665/121419949-d6a94300-c96c-11eb-9989-cd88dd71cbbd.png)

can you help me or suggest me?

![image](https://user-images.githubusercontent.com/6495665/121420027-ed4f9a00-c96c-11eb-9754-5aa0e11a385d.png)

**Comment by madMAx43v3r** [comment]
yeah your CPU doesnt support the bitfield feature... not much I can do about it, it's needed

**Comment by Hardcore-fs** [comment]
The only way is to hand code some replacement assembly then test like they do in the  "blake3" code.
if the instruction is missing, then replace in an alternative routine, but it's a lot of work.

**Comment by Chida82** [comment]
I understand, I would have needed to use an old server with 15k sas

**Comment by Chida82** [comment]
@madMAx43v3r can you show me the first rows of code where the program use bitfield?

**Comment by Hardcore-fs** [comment]
15K SAS is good.
go to the intel website and find their developer tools for specing the cpu
or
https://www.cpuid.com/

I suspect the blake3 is not switching correctly....

**Comment by Chida82** [comment]
@Hardcore-fs i dont realy hunderstand. Do you suggest this? 
https://software.intel.com/content/www/us/en/develop/articles/free-intel-software-developer-tools.html
the other link is a gui that i don't have.

the CPU is Intel Xeon x5450.

flags           : fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ht tm pbe syscall nx lm constant_tsc arch_perfmon pebs bts rep_good nopl cpuid aperfmperf pni dtes64 monitor ds_cpl vmx est tm2 ssse3 cx16 xtpr pdcm dca sse4_1 lahf_lm pti tpr_shadow vnmi flexpriority dtherm

![image](https://user-images.githubusercontent.com/6495665/121772118-c4e5bc80-cb73-11eb-95b0-093cb4f01a20.png)

**Comment by Hardcore-fs** [comment]
you have things like AES missing.
you need to track it down with a debugger to get the op code.
or dig about in teh core dump file.


use "sysctl" to build a debug file then execute within the debugger. gdb
just run it against the built file.

**Comment by Chida82** [comment]
do you have a tutorial for do it?

**Comment by Hardcore-fs** [comment]
just google "debugging"  "gdb".

it's like a 5 minute job to setup & run...
for osx....

1. build the debug version: ./make_debug.sh
2. make sym library: dsymutil /Volumes/null/chia-plotter/build/chia_plot 
3. launch debugger: lldb /Volumes/null/chia-plotter/build/chia_plot 
4. load symbols for debug: target symbols add /Volumes/null/chia-plotter/build/chia_plot.dSYM 
5. and run in dubugger: r
6. report findings about crash point, both  binary & source.
